src/core/quantum_circuit.jl

```diff
@@ -85,6 +85,11 @@ q[2]:───────X────X─────────H──
 ```
 """
 function Base.push!(circuit::QuantumCircuit, gates::AbstractGate...)
+    # convert input from Tuple to Vector
+    return push!(circuit,[gate for gate in gates])
+end
+
+function Base.push!(circuit::QuantumCircuit, gates::Vector{<:AbstractGate})
```

Comment:
This function should be named `append!` to be consistent with Julia's `Base.*` methods.